shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/yaml/config/swapper/mode/ModeConfigurationYamlSwapper.java

```diff
@@ -40,8 +39,10 @@ public YamlModeConfiguration swapToYamlConfiguration(final ModeConfiguration dat
     
     @Override
     public ModeConfiguration swapToObject(final YamlModeConfiguration yamlConfig) {
-        PersistRepositoryConfiguration repositoryConfig = null == yamlConfig.getRepository()
-                ? null : new PersistRepositoryConfigurationYamlSwapperEngine().swapToObject(yamlConfig.getType(), yamlConfig.getRepository());
-        return new ModeConfiguration(yamlConfig.getType(), repositoryConfig, yamlConfig.isOverwrite());
+        if (null == yamlConfig || null == yamlConfig.getRepository()) {
+            return new ModeConfiguration("Memory", null, false);
```

Comment:
Please avoid use type details such as `Memory` in the up level code.
Just use RequiredSPI to resolve it.